source/src/protocols/motifs/LigandDiscoverySearch.cc

```diff
@@ -292,694 +366,1790 @@ utility::vector1<core::conformation::ResidueOP> LigandDiscoverySearch::get_all_r
+			//assign the new cutoff once we hit 100 entries
+			if ( best_placements.size() >= best_pdbs_to_keep && best_pdbs_to_keep != 0 ) {
+				ddg_cutoff_ = std::get<0>(best_placements[0]);
+
+				ms_tr.Debug << "New ddg cutoff is: " << ddg_cutoff_ << std::endl;
+
+			}
+		}
+		//sort passing_placements if keeping only a specified amount
+		if ( best_pdbs_to_keep != 0 ) {
+			best_placements = mergeSort(best_placements);
+		}
+
+		//create pdbs of best scoring poses for pdb
+
+		for ( const auto & best_pose_runner : best_placements ) {
+
+			ms_tr.Debug << "Making pdb file for: " << std::get<2>(best_pose_runner) << std::endl;
+
+			core::io::pdb::dump_pdb( ( std::get<1>(best_pose_runner)), std::get<2>(best_pose_runner));
+		}
+
+
+		ms_tr.Debug << "Number of placements that passed all filters: " << passed_placement_counter << std::endl;
+
+	}
+	return 1;
+}
+
+// @brief A function to reset the working_pose_ back to its original state, as found in original_pose
+//to call this function the least amount of times (and minimize wasted time/resources), this function is only to be called right before a placement is culled before a continue statement after working_pose_ had a ligand added to it
+void LigandDiscoverySearch::reset_working_pose()
+{
+	core::pose::PoseOP original_poseop(original_pose_.clone());
+	//assign the original_poseop to working_pose_
+	working_pose_ = original_poseop;
+}
+
+//function to get a sub-library of motifs from the main library, based on the residue being used (only get for select residue)
+//function may have use outside discover, so public can use it
+protocols::motifs::MotifCOPs LigandDiscoverySearch::get_motif_sublibrary_by_aa(std::string residue_name)
+{
+	static basic::Tracer ms_tr( "LigandDiscoverySearch.get_motif_sublibrary_by_aa", basic::t_info );
+
+	//create temporary motifcops object to hold
+	protocols::motifs::MotifCOPs motif_holder;
+
+	for ( auto motifcop : motif_library_ ) {
+		//collect residue name from motifcop
+		std::string motif_residue_name(motifcop->restype_name1());
+
+		//if residue name matches that of the residue at the working position, add it to the motif_holder
+		if ( motif_residue_name == residue_name ) {
+			motif_holder.push_back(motifcop);
+		}
+	}
+
+	//assign motif_holder contents to motif_library_for_select_residue_
+	motif_library_for_select_residue_ = motif_holder;
+
+
+	ms_tr.Debug << "Created motif sub-library for residue " << residue_name << " with " << motif_library_for_select_residue_.size() << " motifs in it." << std::endl;
+
+
+	return motif_holder;
+}
+
+//create protein_representation_matrix_
+//uses working_pose to make the matrix
+void LigandDiscoverySearch::create_protein_representation_matrix(core::Size & x_shift, core::Size & y_shift, core::Size & z_shift, int & x_bound_int, int & y_bound_int, int & z_bound_int)
+{
+
+	//create tracer to identify points of the run
+	static basic::Tracer ms_tr( "LigandDiscoverySearch.create_protein_representation_matrix", basic::t_info );
```

Comment:
Having a tracer per function might be a bit overkill, unless you think you really need that granularity in control. (Usually a class level assignment of error/warning/info/debug/trace is sufficient -- if you need debug level output for one function in the class, getting debug-level output for all the other functions usually doesn't hurt anything, and may help.)